data-processing/pulsar-flink/2.7.0/pulsar-flink.md

```diff
@@ -0,0 +1,536 @@
+| clientcachesize | 100 | Set the number of cached Pulsar clients. | source, sink |
+| auth-params | null | Set the authentication parameters for Pulsar clients. | source, sink |
+| auth-plugin-classname | null | Set the authentication class name for Pulsar clients.  | source, sink |
+| flushoncheckpoint | true | Write a message to Pulsar topics. | sink |
+| failonwrite | false | When sink error occurs, continue to confirm the message. | sink |
+| polltimeoutms | 120000 | Set the timeout for waiting to get the next message, in unit of milliseconds. | source |
+| failondataloss | true | When data is lost, the operation fails. | source |
+| commitmaxretries | 3 | Set the maximum number of retries when an offset is set for Pulsar messages. | source |
+| scan.startup.mode | null | Set the earliest, latest, and the position where subscribers consume news,. It is a required parameter. | source |
+| enable-key-hash-range | false | Enable the Key-Shared subscription mode. | source |
+| pulsar.reader.* | | For details about Pulsar reader configurations, see [Pulsar reader](https://pulsar.apache.org/docs/en/client-libraries-java/#reader). | source |
+| pulsar.reader.subscriptionRolePrefix | flink-pulsar- | When no subscriber is specified, the prefix of the subscriber name is automatically created. | source |
+| pulsar.reader.receiverQueueSize | 1000 | Set the receive queue size. | source |
+| pulsar.producer.* | | For details about Pulsar producer configurations, see [Pulsar producer](https://pulsar.apache.org/docs/en/client-libraries-java/#producer). | Sink |
+| pulsar.producer.sendTimeoutMs | 30000 | Set the timeout for sending a message, in unit of milliseconds. | Sink |
+| pulsar.producer.blockIfQueueFull | false | The Pulsar producer writes messages. When the queue is full, the method is blocked instead of an exception is thrown. | Sink |
+
+`pulsar.reader.*` and `pulsar.producer.*` specify more detailed configuration of the Pulsar behavior. The asterisk sign (*) is replaced by the configuration name in Pulsar. For details, see [Pulsar reader](https://pulsar.apache.org/docs/en/client-libraries-java/#reader) and [Pulsar producer](https://pulsar.apache.org/docs/en/client-libraries-java/#producer).
+
+In the DDL statement, the sample which is similar to the following is used.
+
+```
+'properties.pulsar.reader.subscriptionRolePrefix' = 'pulsar-flink-',
+'properties.pulsar.producer.sendTimeoutMs' = '30000',
+```
+
+## Authentication configuration
+
+For Pulsar instances configured with authentication, the Pulsar Flink connector can be configured in a similar as the regular Pulsar client.
+
+For `FlinkPulsarSource` and `FlinkPulsarSink`, you can use one of the following ways to set up authentication.
+
+- Set the `Properties` parameter.
+
+  ```java
+  props.setProperty(PulsarOptions.AUTH_PLUGIN_CLASSNAME_KEY, "org.apache.pulsar.client.impl.auth.AuthenticationToken");
+  props.setProperty(PulsarOptions.AUTH_PARAMS_KEY, "token:abcdefghijklmn");
```

Comment:
It should be the ci failure caused by this token, remove the content after the token

Reply:
> It should be the ci failure caused by this token, remove the content after the token

doc is updated accordingly. PTAL